src/internal/ReplaySubject.ts

```diff
@@ -80,6 +96,15 @@ export class ReplaySubject<T> extends Subject<T> {
       subscriber.next(copy[i] as T);
     }
 
+    // Emit values that are synchronously pushed
+    for (let i = 0; i < syncValues.length && !subscriber.closed; i++) {
+      subscriber.next(syncValues[i]);
+    }
+
+    // Swap subscription to the original one
+    syncSubscription.unsubscribe();
```

Comment:
line 73 to 105 should be wrapped in an `if (_buffer.length > 0)` (and maybe moved into a private helper method `replay`)

Reply:
I personally wouldn't add more cyclomatic complexity to avoid doing something that only happens on the initial subscription. When `_buffer.length === 0` this will work just as fine.